backends-common/rabbitmq/src/main/java/org/apache/james/backends/rabbitmq/ReactorRabbitMQChannelPool.java

```diff
@@ -700,26 +652,47 @@ public int getMaxChannel() {
     }
 
     private static final Logger LOGGER = LoggerFactory.getLogger(ReactorRabbitMQChannelPool.class);
-    private static final long MAXIMUM_BORROW_TIMEOUT_IN_MS = Duration.ofSeconds(5).toMillis();
     private static final int MAX_CHANNELS_NUMBER = 3;
     private static final int MAX_BORROW_RETRIES = 3;
-    private static final Duration MIN_BORROW_DELAY = Duration.ofMillis(50);
+    private static final Duration MAX_BORROW_DELAY = Duration.ofSeconds(30);
 
+    private final ConcurrentHashMap<Integer, PooledRef<? extends Channel>> refs = new ConcurrentHashMap<>();
     private final Mono<Connection> connectionMono;
-    private final GenericObjectPool<Channel> pool;
-    private final ConcurrentSkipListSet<Channel> borrowedChannels;
     private final Configuration configuration;
+
+    private final InstrumentedPool<? extends Channel> newPool;
+    private final MetricFactory metricFatcory;
     private Sender sender;
 
-    public ReactorRabbitMQChannelPool(Mono<Connection> connectionMono, Configuration configuration) {
+    public ReactorRabbitMQChannelPool(Mono<Connection> connectionMono, Configuration configuration, MetricFactory metricFatcory) {
         this.connectionMono = connectionMono;
         this.configuration = configuration;
-        ChannelFactory channelFactory = new ChannelFactory(connectionMono, configuration);
+        this.metricFatcory = metricFatcory;
+
+        newPool = PoolBuilder.from(connectionMono
+            .flatMap(this::openChannel))
+            .sizeBetween(1, configuration.maxChannel)
+            .maxPendingAcquireUnbounded()
+            .evictionPredicate((channel, metadata) -> {
+                if (!channel.isOpen()) {
+                    return true;
+                }
+                if (metadata.idleTime() > Duration.ofSeconds(30).toMillis()) {
```

Comment:
```suggestion
                if (metadata.idleTime() > configuration.maxBorrowDelay.toMillis()) {
```

Reply:
No this is different... Time before eviction is different than timeout when acquiring a channel...